web/src/components/countrytable.js

```diff
@@ -327,24 +328,32 @@ const CountryElectricityProductionTable = React.memo(({
 
   const { productionY, exchangeFlagX, exchangeY } = getDataBlockPositions(productionData, exchangeData);
 
+  // Use the whole history to determine the min/max,
+  // fallback on current data
+  const history = useCurrentZoneHistory() || [data];
```

Comment:
Ah I missed this earlier but if `useCurrentZoneHistory` is undefined here, this will always recreate a new array and therefore the `useMemo` below won't really be memoizing properly but will be recalculated whenever any of the props change.

For a further optimization, I'd recommend changing this line to `const history = useCurrentZoneHistory()` and then adding `data` dependency to `useMemo` below and handling it as a fallback separately within its function's body.

But I imagine this is not such a big deal anyway as most of the time `useCurrentZoneHistory()` won't be undefined so up to you whether you want to address it here :wink:

Reply:
Thanks for catching this. Consistency is key so I'll edit it!